File: src/test/java/application/browser/LibraryHelper.java
package application.browser;

import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;

import com.matag.player.PlayerType;

public class LibraryHelper extends AbstractCardContainerHelper {
  private final PlayerType playerType;

  LibraryHelper(MatagBrowser matagBrowser, PlayerType playerType) {
    super(matagBrowser);
    this.playerType = playerType;
  }

  @Override
  protected WebElement containerElement() {
    return matagBrowser.findElement(By.id(playerType.name().toLowerCase() + "-library"));
  }

  public VisibleLibraryHelper getVisibleCardsHelper() {
    return new VisibleLibraryHelper(this.matagBrowser, playerType);
  }
}

File: src/test/kotlin/application/browser/LibraryHelper.kt
package application.browser

import com.matag.player.PlayerType
import org.openqa.selenium.By
import org.openqa.selenium.WebElement
import java.util.*

class LibraryHelper internal constructor(matagBrowser: MatagBrowser?, private val playerType: PlayerType) :
    AbstractCardContainerHelper(matagBrowser) {
    override fun containerElement(): WebElement? {
        return matagBrowser.findElement(By.id(playerType.name.lowercase(Locale.getDefault()) + "-library"))
    }

    val visibleCardsHelper: VisibleLibraryHelper
        get() = VisibleLibraryHelper(this.matagBrowser, playerType)
}
